File: sheet 8 O. Free Ice Cream/main.js
var input = readline().split` `.map(x => +x)
var q = input[0], icecream = input[1];
var left = 0

while (q--) {
    var input = readline().split` `
    var option = input[0], amount = +input[1];
    switch (option) {
        case '+':
            icecream += amount;
            break;
        case '-':
            if (icecream >= amount) {
                icecream -= amount
            } else {
                left++;
            }
    }
}
print(`${icecream} ${left}`)

File: sheet 8 O. Free Ice Cream/main.py
n,x=map(int,input().split())

res=0
for  _ in range(n):
    opt,num=map(str,input().split())
    num=int(num)
    if opt=="+":
        x+=num
    else:
        if num>x:
            res+=1
            continue
        else:
            x-=num
print(x,res)